### PR TITLE
Online/offline info bar

### PR DESCRIPTION
**Pull request by dgboss**
- adds an `InfoBar` component which always displays the viewing date and last updated
- "Last updated" text was changed from the mockups to just be "Updated" due to space limitations
- There are three paths for updating `lastUpdated` in `dataSlice`:
1. Online and new data (prov summaries, hfi and tpi) has been fetched -> `lastUpdated` = now
2. Online and a check for new `runParameters` indicates that new `runParameters` are not available -> `lastUpdated` = now as no new run parameters means no new hfi data
3. Offline and cached data is loaded into redux state -> `lastUpdated` = the lastUpdated date as read from the cached provincial summaries data (which matches hfi and tpi last updated dates)

Closes: #5282 

# Test Links:
[Landing Page](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireCalc](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireCalc bookmark](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)
[SFMS Insights](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/insights)
[Fire Watch](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-watch)
[Weather Toolkit](https://wps-pr-5329-e1e498-dev.apps.silver.devops.gov.bc.ca/weather-toolkit)

### CODE REVIEW

File: mobile/asa-go/src/slices/runParametersSlice.ts [hunk 1]
```diff
@@ -58,92 +60,68 @@ export const {
...
+      dispatch(getRunParametersFailed("Unable to update runParameters from the API."));
+      return;
+    }
+
+    await writeToFileSystem(Filesystem, RUN_PARAMETERS_CACHE_KEY, latestRunParameters, today);
+
+    if (isNil(reduxRunParameters) || stateUpdateRequired(todayKey, tomorrowKey, reduxRunParameters, latestRunParameters)) {
+      dispatch(getRunParametersSuccess({ runParameters: latestRunParameters }));
+    } else {
+      dispatch(setLastUpdated({ lastUpdated: DateTime.now().toISO() }));
```
Comment (conbrad) [inline mobile/asa-go/src/slices/runParametersSlice.ts]: `writeToFileSystem` above uses the module-level today (set at app load) instead of `DateTime.now()` at fetch time that's used here. Should we use the same timestamp for last updated and filesystem writes?
It looks like the same timestamp is used in `mobile/asa-go/src/slices/dataSlice.ts`

File: mobile/asa-go/src/slices/runParametersSlice.ts
Comment (conbrad) [inline mobile/asa-go/src/slices/runParametersSlice.ts:1]: Should we be resetting the loading state on success and failure states?

File: mobile/asa-go/src/App.tsx [hunk 2]
```diff
@@ -280,6 +285,15 @@ const App = () => {
           overflow: "hidden",
         }}
       >
+        <InfoBar
+          status={
```
Comment (conbrad) [inline mobile/asa-go/src/App.tsx:289]: Should we show and error state if any data lookups have failures?

File: mobile/asa-go/src/slices/dataSlice.test.ts [hunk 3]
```diff
@@ -273,11 +276,17 @@ describe("data reducer", () => {
     expect(nextState.error).toBe(error);
   });
 
+  it("should handle setLastUpdated", () => {
```
Comment (conbrad) [inline mobile/asa-go/src/slices/dataSlice.test.ts:280]: Can we also test `lastUpdated` from the cache path?

File: mobile/asa-go/src/components/InfoBar.test.tsx
Comment (conbrad) [inline mobile/asa-go/src/components/InfoBar.test.tsx:1]: Can we also add a test for the empty string `statusText=""` We have a falsy check in the component.

File: mobile/asa-go/src/components/InfoBar.tsx
Comment (brettedw) [inline mobile/asa-go/src/components/InfoBar.tsx:1]: Image

Everything doesn't quite fit on very small screens like the SE for Offline mode. I think you may be able to use the `xs` and/or `sm` breakpoints to adjust for these small screens

Comment (dgboss) [reply mobile/asa-go/src/components/InfoBar.tsx:1]: Consulted with UX and the suggested solution is to abbreviate Updated -> Upd for all screen sizes for consistency. I'm also now using just two spans for content so in the event someone alters display settings the string will wrap nicely to the second line (screenshot taken before abbreviated Updated).

image

Done in https://github.com/bcgov/wps/pull/5329/changes/cc9a51bd9c9404c7e9ecdf8d3558ebbc677bcc1d.

File: mobile/asa-go/src/slices/runParametersSlice.ts
Comment (dgboss) [reply mobile/asa-go/src/slices/runParametersSlice.ts:1]: We do reset loading state on success and failure. We can probably get rid of the runParameter loading state as I don't think it is used anywhere? Is that what you're getting at?

Comment (conbrad) [reply mobile/asa-go/src/slices/runParametersSlice.ts:1]: Yeah my bad, it's unused, should we use it?

Comment (dgboss) [reply mobile/asa-go/src/slices/runParametersSlice.ts:1]: I think I'll just remove it for now.

File: mobile/asa-go/src/App.tsx (hunk 2 above)
Comment (dgboss) [reply mobile/asa-go/src/App.tsx:289]: The `Viewing` date should always be there. A `n/a` displays if `lastUpdated` is null.  I think the transient alerts that UX has designed make more sense for displaying data lookup failures.

File: mobile/asa-go/src/components/InfoBar.test.tsx
Comment (dgboss) [reply mobile/asa-go/src/components/InfoBar.test.tsx:1]: Added in https://github.com/bcgov/wps/pull/5329/changes/a0fff950d79378cf46ca3e0a27ebaf0006c64bc9

File: mobile/asa-go/src/slices/dataSlice.test.ts (hunk 3 above)
Comment (dgboss) [reply mobile/asa-go/src/slices/dataSlice.test.ts:280]: Added in https://github.com/bcgov/wps/pull/5329/changes/a0fff950d79378cf46ca3e0a27ebaf0006c64bc9

File: mobile/asa-go/src/slices/runParametersSlice.ts
Comment (dgboss) [reply mobile/asa-go/src/slices/runParametersSlice.ts:1]: Removed in https://github.com/bcgov/wps/pull/5329/changes/a0fff950d79378cf46ca3e0a27ebaf0006c64bc9

File: mobile/asa-go/src/slices/runParametersSlice.ts (hunk 1 above)
Comment (dgboss) [reply mobile/asa-go/src/slices/runParametersSlice.ts]: Fixed in https://github.com/bcgov/wps/pull/5329/changes/a0fff950d79378cf46ca3e0a27ebaf0006c64bc9

**Review by conbrad — APPROVED** [review]

**Review by brettedw — APPROVED** [review]
Looks great!

**Review by conbrad — APPROVED** [review]
